### PR TITLE
[BEAM-3654] Port FilterExamplesTest off DoFnTester

### PR DESCRIPTION
**Pull request by SokolovMS**
This PR removes the now deprecated DoFnTester from the FilterExamplesTest. All tests in FilterExamplesTest now use TestPipeline.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

File: examples/java/src/test/java/org/apache/beam/examples/cookbook/FilterExamplesTest.java [hunk 1]
```diff
@@ -54,31 +57,35 @@
...
 
   @Test
-  public void testProjectionFn() throws Exception {
-    DoFnTester<TableRow, TableRow> projectionFn = DoFnTester.of(new ProjectionFn());
-    List<TableRow> results = projectionFn.processBundle(ROWS_ARRAY);
-    Assert.assertThat(results, CoreMatchers.hasItem(outRow1));
-    Assert.assertThat(results, CoreMatchers.hasItem(outRow2));
-    Assert.assertThat(results, CoreMatchers.hasItem(outRow3));
+  @Category(ValidatesRunner.class)
+  public void testProjectionFn1() {
```
Comment (aromanenko-dev) [inline examples/java/src/test/java/org/apache/beam/examples/cookbook/FilterExamplesTest.java]: Any reason that the method name ends with "1"? I'd prefer to keep old name.

File: examples/java/src/test/java/org/apache/beam/examples/cookbook/FilterExamplesTest.java [hunk 2]
```diff
@@ -54,31 +57,35 @@
...
   }
 
   @Test
-  public void testFilterSingleMonthDataFn() throws Exception {
-    DoFnTester<TableRow, TableRow> filterSingleMonthDataFn =
-        DoFnTester.of(new FilterSingleMonthDataFn(7));
-    List<TableRow> results = filterSingleMonthDataFn.processBundle(PROJROWS_ARRAY);
-    Assert.assertThat(results, CoreMatchers.hasItem(outRow2));
+  @Category(ValidatesRunner.class)
+  public void testFilterSingleMonthDataFn1() {
```
Comment (aromanenko-dev) [inline examples/java/src/test/java/org/apache/beam/examples/cookbook/FilterExamplesTest.java]: Any reason that the method name ends with "1"? I'd prefer to keep old name.

**Comment by aromanenko-dev** [comment]
Retest this please

**Comment by timrobertson100** [comment]
Retest this please

**Comment by SokolovMS** [comment]
Applied remarks, added issue ID to squashed commit

**Comment by aromanenko-dev** [comment]
Retest this please

**Comment by aromanenko-dev** [comment]
LGTM

**Pull request merged**